agent/predictions/ratelimit.py

```
@@ -0,0 +1,49 @@
+    "bet": "10/minute",           # Lower for write operations
+    "cashout": "10/minute",
+    "sync": "2/minute",           # Very low for admin operations
+    "leaderboard": "30/minute",
+}
+
+def get_rate_limit(endpoint: str) -> str:
+    """Get rate limit for an endpoint."""
+    return RATE_LIMITS.get(endpoint, RATE_LIMITS["default"])
+
+
+async def rate_limit_exceeded_handler(request: Request, exc: RateLimitExceeded):
```

Ruff (0.14.10)

Unused function argument: `request`

(ARG001)